driver_cpl/driver/cesm_comp_mod.F90

```diff
@@ -3679,6 +3723,33 @@ subroutine cesm_run()
       endif
 
       !----------------------------------------------------------
+      !| RUN ESP MODEL
+      !----------------------------------------------------------
+      if (esp_present .and. esprun_alarm) then
+         call component_run(Eclock_e, esp, esp_run, infodata, &
+              comp_prognostic=esp_prognostic, comp_num=comp_num_esp, &
+              timer_barrier= 'CPL:ESP_RUN_BARRIER', timer_comp_run='CPL:ESP_RUN', &
+              run_barriers=run_barriers, ymd=ymd, tod=tod,comp_layout=esp_layout)
+      endif
+
+      !----------------------------------------------------------
+      !| RESUME (read restart) if signaled
+      !----------------------------------------------------------
+      call seq_infodata_GetData(infodata, cpl_resume=drv_resume)
+      if (len_trim(drv_resume) > 0) then
+        if (iamroot_CPLID) then
+          write(logunit,103) subname,' Reading restart (resume) file ',trim(drv_resume)
+          call shr_sys_flush(logunit)
+        end if
+        if (iamin_CPLID) then
+          call seq_rest_read(drv_resume, infodata,                            &
+               atm, lnd, ice, ocn, rof, glc, wav, esp,                        &
+               fractions_ax, fractions_lx, fractions_ix, fractions_ox,        &
+               fractions_rx, fractions_gx, fractions_wx)
+        end if
+      end if
+
+      !----------------------------------------------------------
```

Comment:
ESP needs to run after any model (including the coupler) has written a restart (pause) file.